### ISSUE TITLE
questions about extracting titles from cocina ... 

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/cocina-models#657]
The questions below arose as I tried to work on improving title indexing for Argo. 

@arcadiafalcone.   Please ping me if you have any questions, or if you are done and want to notify me.  :-P

1. when there is no "status" of primary on a title, how do we know which one to choose among many?  
2. when the primary title is a parallel title, should both value flavors be a part of the title value, or can the "second" parallel value (in order encountered) be considered some flavor of additional title?
3. when there is no primary title, should the uniform title be the next choice?
4.  for uniform titles (where type="uniform"), should the associated name be a part of the title, and precede the "title" portion?   Should all pieces of a structuredValue for the associated name be included? See https://argo.stanford.edu/view/druid:bb022pc9382 for an example.
5. based on some "research," I determined that this is the correct order of structuredValue pieces:  nonsorting, main, subtitle, part names and numbers. Are there any other pieces that might be encountered in a structured title?  

Please add any / correct any specs in sul-dlss/cocina-models/blob/main/spec/cocina/models/builders/title_builder_spec.rb in a PR;  

Please add any missing wrinkles in testing to the same file (can be in same PR) and notify me after.

**Comment by arcadiafalcone** [comment]
For 1, 2, and 3 - it depends on the purpose of the indexing. If it's necessary to select a primary title in the absence of one with `status: primary` primary title, use the first title. If the title contains a `parallelValue`, all values of that property should be treated equally. 4 is correct on both counts. There are exceptions to the order given in 5 but they are rare. In general `structuredValue` should be concatenated in the order given regardless of type. All valid types for title parts are defined here for validation: https://github.com/sul-dlss/cocina-models/blob/main/docs/description_types.md#title-part-types-for-structured-value.

**Comment by ndushay** [comment]
Total records in Argo: 6,615,679
Records with `sw_display_title_tesim`:  6,615,726
Records with folio ids:  4,083,131
Google Books:  2,992,442

**Comment by ndushay** [comment]
Re: uniform titles in SDR:

Arcadia and I just discussed whether it is worth the effort in index the "associated name" as part of a Solr title field.

#### Original Notion:   

- Think of a music context (there may be other contexts):  searching for a title only, like "Sonata No. 14" will have terrible precision.
- In SearchWorks, there is the notion of an "author title" search to address this.  [Example in SearchWorks](https://searchworks.stanford.edu/?q=%22Mozart%2C+Wolfgang+Amadeus%2C+1756-1791.+Sonatas%2C+piano%2C+K.+570%2C+B%E2%99%AD+major%22&search_field=author_title) If you hover over the link for "Sonatas, piano, K570, Bb major", it smushes the primary contributor with the uniform title so clicking the link is searching "Mozart Wolfgang Amadeus 1756-1791 Sonatas piano K. 570 ..."   This is GREAT for classical music folks looking for things entitled "Symphony" or "Sonata" or "Etude" etc.  - Not sure if there is another use case.

#### For Argo:
- we might have a uniform title in cocina like this (from bb001sq9051):
    ```json
        {
          "value":"Printemps (Suite); arranged",
          "type":"uniform",
          "uri":"http://id.loc.gov/authorities/names/no97047088",
          "note": [
            {
              "structuredValue": [
                {
                  "value":"Debussy, Claude",
                  "type":"name"
                },
                {
                  "value":"1862-1918",
                  "type":"life dates"
                }
              ],
              "type":"associated name"
            }
          ]
        }
    ```
- But we also have this contributor
    ```json
    "contributor": [
        {
            "name": [
            {
                "structuredValue": [
                    {"value":"Debussy, Claude","type":"name"},
                    {"value":"1862-1918","type":"life dates"}
                ],
               "uri":"http://id.loc.gov/authorities/names/n79132137"
            }
          ],
        ...
        }
    ```
Which is indexed as a contributor.  And all of these values, including the note in the title, are put in the catch all field, so they are searched.
- Arcadia did a search in Argo for (author) "title in quotes" and found the music reference she was looking for as the first search result.   This implies that we have this sort of work around, which will probably suit Argo user needs just fine.  (How much music is in the SDR?  How many SDR objects have descriptive metadata that includes a uniform title and an associated name?)

#### Conclusion on Uniform Titles
All of the above has led Arcadia and I to decide NOT to worry about putting the associated name in a TITLE field for searching in Argo.   Thus making the original description of this ticket, point 4, "No".

**Comment by ndushay** [comment]
Closing this.

1. when there is no "status" of primary on a title, how do we know which one to choose among many?  DONE.  cocina-models method uses algorithm already present in TitleBuilder, which picks the first one.  (which matches Arcadia's answer)
2. when the primary title is a parallel title, should both value flavors be a part of the title value, or can the "second" parallel value (in order encountered) be considered some flavor of additional title?  DONE.  cocina-models `main_title` and `full_title` and `additional_title` methods get all values from parallel titles.
3. when there is no primary title, should the uniform title be the next choice?  DONE.  The answer is "if it's the first title yes.  Otherwise no.  See 1."
4.  for uniform titles (where type="uniform"), should the associated name be a part of the title, and precede the "title" portion?   Should all pieces of a structuredValue for the associated name be included? See https://argo.stanford.edu/view/druid:bb022pc9382 for an example.  DONE.  See previous comment -- we're not doing this.
5. based on some "research," I determined that this is the correct order of structuredValue pieces:  nonsorting, main, subtitle, part names and numbers. Are there any other pieces that might be encountered in a structured title?   DONE.  My approach was wrong, and I backed it out.
